### PR TITLE
Blobstore clients supports PUT headers

### PR DESCRIPTION
**Pull request by mvach**
### What is this change about?

When putting blobs via signed urls to Azure Blobstore, then the mandatory header 'x-ms-blob-type' has to be set to value 'blockblob'. Therefore we add a mechanism to support PUT headers.

We are doing it just the same why like encryption headers can be set for GCS.

### CODE REVIEW

**Comment by mvach** [comment]
Hi @jpalermo,
we did not test this PR right now but just want to double check if the approach is right from you perspective.
So you feedback is very welcome :-)

**Comment by jpalermo** [comment]
I haven't looked at it super closely, but I wonder if we can just combine all the headers in the blobstore client classes and rename `signed_url_encryption_header` to just `signed_url_headers` and let the blobstore object determine which headers should be included.

**Comment by mvach** [comment]
After incorporating the feedback we started validating the PR in our local test environment.

**Comment by mvach** [comment]
We tested this PR successfully on our azure landscape.

**Comment by mvach** [comment]
Hi @jpalermo, hi @aramprice,
a review would be very nice.

**Review by fmoehler — APPROVED** [review]

File: src/bosh-director/lib/bosh/blobstore_client/azurestoragecli_blobstore_client.rb [hunk 1]
```diff
@@ -40,11 +40,8 @@ def initialize(options)
     def redacted_credential_properties_list
       %w[account_key]
     end
```
Comment (Malsourie) [inline src/bosh-director/lib/bosh/blobstore_client/azurestoragecli_blobstore_client.rb:42]: Empty line between methods

**Review by Malsourie — APPROVED** [review]

File: src/bosh-director/spec/unit/deployment_plan/stages/package_compile_stage_spec.rb [hunk 2]
```diff
@@ -353,9 +353,7 @@ def compile_package_with_url_encrypt_stub(args)
       end
     end
 
-    context 'when none of the packages are compiled' do
-      let(:instance_groups_to_compile) { [@j_dea, @j_router] }
-
+    shared_examples 'compiles all packages' do |compilation_method, compilation_method_stub|
```
Comment (aramprice) [inline src/bosh-director/spec/unit/deployment_plan/stages/package_compile_stage_spec.rb:356]: The changes from here down to line 384 seem unrelated to headers. Is this an intentional change?

**Comment by mvach** [comment]
Hi @aramprice,
if you are fine with the PR, then an approval would be nice.

File: src/bosh-director/lib/bosh/blobstore_client/azurestoragecli_blobstore_client.rb (hunk 1 above)
Comment (mvach) [reply src/bosh-director/lib/bosh/blobstore_client/azurestoragecli_blobstore_client.rb:42]: Hi @Malsourie,
thx for the hint. I changed it.

**Review by jpalermo — APPROVED** [review]
👍
